### ISSUE TITLE
[Question] How can I reference the value of the `AutoCompleteInput`?

### ISSUE THREAD

**Issue by SpencerKaiser** [anubra266/choc-autocomplete#111]
I have a few things I'd like to tweak for my use case, so I need access to the current value of the input. Ideally I'd have access to it so I could change the empty state to include a `Spinner` as I wait for options to be populated from an async call.

**Comment by anubra266** [comment]
@SpencerKaiser could you explain more what you're trying to do, don't really understand

**Comment by SpencerKaiser** [comment]
@anubra266 thanks for getting back to me so quickly and thank you for the AWESOME component... it was perfect for what I needed 🙂 

So here's what I'm trying to do:
``` typescript
<AutoComplete
              openOnFocus
              emptyState={
                <Box ml={3}>
                  { inputHasValue ? 'Loading' : 'Start typing your address for suggestions' }
                </Box>
              }
              onSelectOption={(selectedOption) => {
                formik.values.address = selectedOption.item.value;
                formik.validateForm();
              }}
            >
```

Does that make sense? Basically I have a few places where knowing if the user has typed info into the input is meaningful

**Comment by anubra266** [comment]
But, the emptyState will never render when the input is empty though @SpencerKaiser

**Comment by anubra266** [comment]
But I understand, you want access to the value of the Input.

**Comment by anubra266** [comment]
@SpencerKaiser I just release [v4.15.2](https://github.com/anubra266/choc-autocomplete/releases/tag/v4.15.2).
I forwarded a `ref` to the Input, so you can add a `ref` to `AutoCompleteInput` then, get the value with `inputRef?.current?.value`

**Comment by SpencerKaiser** [comment]
When you initially click on the input, the empty state displays. The workaround I figured out (just maintaining my own boolean and setting it when it changes and after I fetch select items) works, here's a demo:

https://user-images.githubusercontent.com/6445731/150657524-1717c9d5-49a3-4f78-81c3-9b71e669dfec.mov

Thank you for adding the `ref`! That sounds like a perfect solution and it's one less thing I need to keep track of state for!

You're awesome, thank you so much 🎉

**Comment by SpencerKaiser** [comment]
@anubra266 that worked perfectly 🤌 

For anyone else who stumbles upon this, here's what I did after updating the package to the new version posted above:
``` typescript
// Somewhere inside your component
const inputRef = React.useRef<HTMLInputElement>(null);

// ...
<AutoComplete
  openOnFocus
  emptyState={
    <Box textAlign="center">
      {inputRef?.current?.value ? (
        <Spinner />
      ) : (
        <Box>Start typing your address for suggestions</Box>
      )}
    </Box>
  }
>
  <AutoCompleteInput
    ref={addressInput}
  />
```

**Comment by IamMrandrew** [comment]
@anubra266  Is it possible to **get notified when the input value is changed**? Like the `onChange` property on the `input` tag, not the existing `onChange` when the autocomplete value(s) changes.

I tried to put ref into it using `useRef()` but clearly it won't work as it won't notify when `ref.current.value` has changed.

I also tried to use the `useCallback` for the ref. But surprisingly, it does not work as expected.

```jsx
const onRefChange = useCallback((ref) => {
    console.log(ref);
})

<AutoCompleteInput ref={onRefChange} />
```

For more details, I am working on a Google Map with `use-places-autocomplete` package. The `suggestions.data` array is retrieved based on the `value`. And I am expected to update the `value` with `setValue` on input change.

```jsx
 const {
    ready,
    value,
    suggestions: { status, data },
    setValue,
    clearSuggestion,
  } = usePlacesAutocomplete({
    requestOptions: {
      location: { lat: () => 22.396427, lng: () => 114.109497 },
      radius: 100 * 1000,
    },
  });
```

It will add flexibility if I am able to not only get the `value` but also able to monitor the `value` from the input. ✨
